common/src/main/java/com/microsoft/identity/common/internal/cache/MsalAccessTokenCacheItem.java

```diff
@@ -0,0 +1,71 @@
+package com.microsoft.identity.common.internal.cache;
+
+import com.google.gson.Gson;
+import com.google.gson.annotations.SerializedName;
+import com.microsoft.identity.common.Account;
+import com.microsoft.identity.common.internal.providers.azureactivedirectory.AzureActiveDirectoryAccount;
+import com.microsoft.identity.common.internal.providers.azureactivedirectory.AzureActiveDirectoryTokenResponse;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationRequest;
+import com.microsoft.identity.common.internal.providers.oauth2.OAuth2Strategy;
+import com.microsoft.identity.common.internal.providers.oauth2.TokenResponse;
+import com.microsoft.identity.common.internal.util.EncodingUtil;
+
+class MsalAccessTokenCacheItem extends BaseMsalTokenCacheItem {
+
+    @SerializedName("authority")
+    String mAuthority;
+
+    @SerializedName("access_token")
+    String mAccessToken;
+
+    @SerializedName("expires_on")
+    long mExpiresOn;
+
+    @SerializedName("scope")
+    String mScope;
+
+    @SerializedName("token_type")
+    String mTokenType;
+
+    @SerializedName("id_token")
+    String mRawIdToken;
+
+    transient String mUserIdentifier;
+
+    public MsalAccessTokenCacheItem(OAuth2Strategy oAuth2Strategy, AuthorizationRequest request, TokenResponse response) {
+        super(oAuth2Strategy, request, response);
+        mAuthority = ""; // TODO where can I get this?
+        mScope = request.getScope();
+        mAccessToken = response.getAccessToken();
+        if (response instanceof AzureActiveDirectoryTokenResponse) {
+            mExpiresOn = ((AzureActiveDirectoryTokenResponse) response).getExpiresOn().getTime();
+        }
+        mTokenType = response.getTokenType();
+        mRawIdToken = response.getIdToken();
+        final Account account = oAuth2Strategy.createAccount(response);
+        if (account instanceof AzureActiveDirectoryAccount) {
+            final AzureActiveDirectoryAccount aadAcct = (AzureActiveDirectoryAccount) account;
+            mUserIdentifier = ""; // TODO which unique identifier should I use?
```

Comment:
I need to dig around in MSAL and figure out which of the many IDs I need to use.... Or do I need to do some trickery around `getCacheIdentifiers()`? This is unclear to me...

Reply:
I added a comment to the PR to address this.  For MSAL i'd suggest using the getUniqueIdentifier abstract method of account... since they are not interested in leveraging multiple cache identifiers at this point.